File: Array&String/HashTable/CPP/destination_city.cpp
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_map<string, string> d;
        string dest = paths[0][0];
        for (vector<string> p : paths) d[p[0]] = p[1];
        while (d.count(dest) == 1) dest = d[dest];
        return dest;
    }
};

File: Array&String/HashTable/Python/destination_city.py
from collections import defaultdict

class Solution:
    def destCity(self, paths: list[list[str]]) -> str:
        d = defaultdict(str)
        dest = paths[0][0]
        for p in paths:
            d[p[0]] = p[1]
        while d.get(dest):
            dest = d.get(dest)
        return dest